app.js

```diff
@@ -1,57 +1,15 @@
+import {
+  createRemoteJWKSet,
+  jwtVerify,
+} from 'https://cdnjs.cloudflare.com/ajax/libs/jose/4.14.4/index.bundle.min.js';
+
 (function () {
-  function b64DecodeUnicode(str) {
-    return decodeURIComponent(
-      atob(str).replace(/(.)/g, function (m, p) {
-        var code = p.charCodeAt(0).toString(16).toUpperCase();
-        if (code.length < 2) {
-          code = '0' + code;
-        }
-        return '%' + code;
-      })
+  async function validateSignature(token) {
+    const JWKS = createRemoteJWKSet(
+      new URL('https://app.wellnesstogether.grnspace.ca/oauth2/jwks/')
     );
-  }
-
-  function base64UrlDecode(str) {
-    var output = str.replace(/-/g, '+').replace(/_/g, '/');
-    switch (output.length % 4) {
-      case 0:
-        break;
-      case 2:
-        output += '==';
-        break;
-      case 3:
-        output += '=';
-        break;
-      default:
-        throw 'Illegal base64url string!';
-    }
-
-    try {
-      return b64DecodeUnicode(output);
-    } catch (err) {
-      return atob(output);
-    }
-  }
-
-  function InvalidTokenError(message) {
-    this.message = message;
-  }
-
-  InvalidTokenError.prototype = new Error();
-  InvalidTokenError.prototype.name = 'InvalidTokenError';
-
-  function jwtDecode(token, options) {
-    if (typeof token !== 'string') {
-      throw new InvalidTokenError('Invalid token specified');
-    }
-
-    options = options || {};
-    var pos = options.header === true ? 0 : 1;
-    try {
-      return JSON.parse(base64UrlDecode(token.split('.')[pos]));
-    } catch (e) {
-      throw new InvalidTokenError('Invalid token specified: ' + e.message);
-    }
+    const { payload } = await jwtVerify(token, JWKS);
+    return payload;
```

Comment:
I wonder if we can print to the console or the html dom some info about the verification 🤔.

Reply:
Ooh yeah that would be good. Incoming...

Reply:
Like this:
<img width="504" alt="image" src="https://github.com/grnspace/wtc-oidc-client/assets/16126542/ecfccde0-db96-4f6e-8f45-bc981a414130">